tests/util/Component.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+const assert = require('assert');
+const eslint = require('eslint');
+const Components = require('../../lib/util/Components');
+
+const ruleTester = new eslint.RuleTester({
+  parserOptions: {
+    ecmaVersion: 2018,
+    sourceType: 'module',
+    ecmaFeatures: {
+      jsx: true,
+    },
+  },
+});
+
+describe('Components', () => {
+  describe('static detect', () => {
+    function makeTest(name, test, done) {
+      return it(name, () => {
```

Comment:
i'm not much of a fan of dynamically created tests like this; could we make the `it`s all hardcoded, and do the rest dynamically?

even better might be to avoid using `it`s at all, and just use ruletester with a dynamically built-up array of test cases?

Reply:
I feel the same way, but this was the most expedient.
I'll see about teasing this apart a little bit.